src/main/java/org/jabref/gui/customentrytypes/CustomEntryTypeDialogViewModel.java

```diff
@@ -126,7 +120,26 @@ public String toString() {
     public void addNewField() {
         Field field = newFieldToAdd.getValue();
         FieldViewModel model = new FieldViewModel(field, true, FieldPriority.IMPORTANT);
-        this.selectedEntryType.getValue().addField(model);
+
+        boolean fieldExists = false;
+
+        // create list with all the entry's fields
+        ObservableList<FieldViewModel> entryFields = this.selectedEntryType.getValue().fields();
+
+        // compare every entry field name with the user field name in order to find out if any of them has the same one. If so, show warning.
+        for (FieldViewModel fieldViewModel : entryFields) {
```

Comment:
This can be simplified
`entryFields.stream().anyMatch(fieldViewModel -> fieldViewModel.fieldName().getValue().equals(...)`
See https://www.baeldung.com/java-8-streams-introduction#1-iterating